packages/core/addon/models/bard-request-v2/request.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import attr from 'ember-data/attr';
+import Fragment from 'ember-data-model-fragments/fragment';
+import { fragmentArray } from 'ember-data-model-fragments/attributes';
+import { validator, buildValidations } from 'ember-cp-validations';
+import { getDefaultDataSourceName } from 'navi-data/utils/adapter';
+
+const Validations = buildValidations({
+  filters: [
+    validator('has-many'),
+    validator('collection', {
+      collection: true,
+      message: 'Filters must be a collection'
+    })
+  ],
+  columns: [
+    validator('has-many'),
+    validator('collection', {
+      collection: true,
+      message: 'Columns must be a collection'
+    }),
+    validator('length', {
+      min: 1,
+      message: 'At least one column should be selected'
+    })
+  ],
+  table: validator('presence', {
+    presence: true,
+    message: 'Table is invalid or unavailable'
+  }),
+  sort: [
+    validator('has-many'),
+    validator('collection', {
+      collection: true,
+      message: 'Sort must be a collection'
+    })
+  ],
+  responseFormat: validator('presence', {
+    presence: true,
+    message: 'Response format cannot be empty'
+  })
+});
+
+export default Fragment.extend(Validations, {
+  filters: fragmentArray('bard-request-v2/fragments/filter', { defaultValue: [] }),
+  columns: fragmentArray('bard-request-v2/fragments/column', { defaultValue: [] }),
+  table: attr('string'),
+  sort: fragmentArray('bard-request-v2/fragments/sort', { defaultValue: [] }),
+  limit: attr('number'),
+  requestVersion: attr('string', { defaultValue: '2.0' }),
+  dataSource: attr('string', { defaultValue: getDefaultDataSourceName() }),
+  responseFormat: attr('string', { defaultValue: 'json' }),
```

Comment:
Do we need this, is this part of the spec?

Reply:
I wasn't sure but looks like it can be removed

Reply:
We can remove